Add spin animation to pipelinerun log tabs

rottencandy opened the pull request:
**Fixes**: 
https://issues.redhat.com/browse/ODC-1857
<!-- For e.g Fixes: https://issues.redhat.com/browse/ODC-XXX -->

**Analysis / Root cause**: 
There is inconsistency in status animations when a pipeline is running.
<!-- Briefly describe analysis of US/Task or root cause of Defect -->

**Solution Description**: 
Add spin animation to task tabs in pipelinerun logs.
<!-- Describe your code changes in detail and explain the solution -->

**Screen shots / Gifs for design review**: 
![anim](https://user-images.githubusercontent.com/20013884/90155383-dec63f80-dda8-11ea-9c44-b5364c88af76.gif)

@openshift/team-devconsole-ux 
<!-- If change affects UI in any way, tag @openshift/team-devconsole-ux and add screenshots/gifs  -->

**Unit test coverage report**: 
Unchanged

**Browser conformance**: 
<!-- To mark tested browsers, use [x] -->
- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

andrewballantyne commented:
@rottencandy You seem to have a whitespace issue in your gif. We definitely want that gap between the icon and the text, spinning or not.

rottencandy commented:
> @rottencandy You seem to have a whitespace issue in your gif. We definitely want that gap between the icon and the text, spinning or not.

Fixed and updated above gif. PTAL.

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/detail-page-tabs/PipelineRunDetails.tsx:
```diff
@@ -15,6 +15,7 @@ import PipelineRunVisualization from './PipelineRunVisualization';
 import TriggeredBySection from './TriggeredBySection';
 
 import './TriggeredBySection.scss';
+import { StatusIcon } from '../../pipelines/detail-page-tabs/pipeline-details/StatusIcon';
```
Nit: Sort imports please

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/detail-page-tabs/PipelineRunDetails.tsx:
```diff
@@ -27,6 +28,7 @@ export const PipelineRunDetails: React.FC<PipelineRunDetailsProps> = ({ obj: pip
     unfilteredResources
       ?.filter(({ resourceRef }) => !!resourceRef)
       .map((resource) => resource.resourceRef.name) || [];
+  const pipelineRunStatus = pipelineRunFilterReducer(pipelineRun);
```
```suggestion
  const pipelineRunStatus = pipelineRunStatus(pipelineRun);
```
That way you get a null-able value.

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/detail-page-tabs/PipelineRunDetails.tsx:
```diff
@@ -40,7 +42,14 @@ export const PipelineRunDetails: React.FC<PipelineRunDetailsProps> = ({ obj: pip
             <dt>Status</dt>
             <dd>
-              <Status status={pipelineRunFilterReducer(pipelineRun)} />
+              {pipelineRunStatus !== '-' ? (
+                <StatusIconAndText
+                  title={pipelineRunStatus}
+                  icon={<StatusIcon status={pipelineRunStatus} />}
+                />
+              ) : (
+                '-'
+              )}
```
```suggestion
                <StatusIconAndText
                  title={pipelineRunStatus}
                  icon={<StatusIcon status={pipelineRunStatus} />}
                />
```
Making this much cleaner.

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/detail-page-tabs/PipelineRunLogs.tsx:
```diff
@@ -3,14 +3,14 @@ import * as _ from 'lodash';
 import LogsWrapperComponent from '../logs/LogsWrapperComponent';
 import { getDownloadAllLogsCallback } from '../logs/logs-utils';
 import './PipelineRunLogs.scss';
+import { StatusIcon } from '../../pipelines/detail-page-tabs/pipeline-details/StatusIcon';
```
Nit: Sort imports

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/list-page/PipelineRunRow.tsx:
```diff
@@ -11,10 +11,12 @@ import { PipelineRunModel } from '../../../models';
 import LinkedPipelineRunTaskStatus from '../status/LinkedPipelineRunTaskStatus';
 import { ResourceKebabWithUserLabel } from '../triggered-by';
 import { tableColumnClasses } from './pipelinerun-table';
+import { StatusIcon } from '../../pipelines/detail-page-tabs/pipeline-details/StatusIcon';
```
Nit: Sort Imports

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/list-page/PipelineRunRow.tsx:
```diff
@@ -11,10 +11,12 @@ import { PipelineRunModel } from '../../../models';
 const pipelinerunReference = referenceForModel(PipelineRunModel);
 
 const PipelineRunRow: RowFunction<PipelineRun> = ({ obj, index, key, style }) => {
+  const pipelineRunStatus = pipelineRunFilterReducer(obj);
```
Same here, update with `pipelineRunStatus` instead.

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/pipeline-details/StatusIcon.tsx:
```diff
@@ -9,9 +9,11 @@ import {
   SyncAltIcon,
 } from '@patternfly/react-icons';
 import { getRunStatusColor, runStatus } from '../../../../utils/pipeline-augment';
+import * as classNames from 'classnames';
```
Nit: Should be line 2 on imports

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelines/list-page/PipelineRow.tsx:
```diff
@@ -9,11 +9,13 @@ import { PipelineModel, PipelineRunModel } from '../../../models';
 const pipelinerunReference = referenceForModel(PipelineRunModel);
 
 const PipelineRow: RowFunction<Pipeline> = ({ obj, index, key, style }) => {
+  const pipelineStatus = pipelineFilterReducer(obj);
```
Same status function change here.

rottencandy commented:
@andrewballantyne updated the PR. The changes only update the logs tabs now.
![anim](https://user-images.githubusercontent.com/20013884/90155502-074e3980-dda9-11ea-8aaa-17cd0152660b.gif)

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/detail-page-tabs/PipelineRunLogs.tsx:
```diff
@@ -3,7 +3,7 @@ import * as _ from 'lodash';
 import { Link } from 'react-router-dom';
 import { Nav, NavItem, NavList } from '@patternfly/react-core';
-import { StatusIcon } from '@console/shared';
+import { ColoredStatusIcon } from '../../pipelines/detail-page-tabs/pipeline-details/StatusIcon';
```
Nit: Sort import... we really should [code up that lint](https://issues.redhat.com/browse/ODC-3670) to avoid these kind of comments.

andrewballantyne commented:
/hold cancel

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelineruns/detail-page-tabs/PipelineRunLogs.tsx:
```diff
@@ -3,13 +3,13 @@ import * as _ from 'lodash';
 import { pipelineRunFilterReducer } from '../../../utils/pipeline-filter-reducer';
 import { PipelineRun } from '../../../utils/pipeline-augment';
 import { PipelineRunModel } from '../../../models';
 import LogsWrapperComponent from '../logs/LogsWrapperComponent';
 import { getDownloadAllLogsCallback } from '../logs/logs-utils';
+import { ColoredStatusIcon } from '../../pipelines/detail-page-tabs/pipeline-details/StatusIcon';
```
```suggestion
import { ColoredStatusIcon } from '../../pipelines/detail-page-tabs/pipeline-details/StatusIcon';
import LogsWrapperComponent from '../logs/LogsWrapperComponent';
import { getDownloadAllLogsCallback } from '../logs/logs-utils';
```
Distance from source.

rottencandy replied:
Sorry, this really wasn't clear to me at first.
